### DISCUSSION TITLE
How to set source preference from which the answer originated

### DISCUSSION THREAD

**Discussion by trepont** [FlowiseAI/Flowise#475] [Q&A]
We have developed Flowwiese with multiple chains/functions, including documents, Google, forums, and more. One challenge we encountered was determining the priority of each source when fetching answers. For example, if we have configured PDF documents, forums, and Google, and we receive an answer to a question, it becomes difficult to identify the specific source from which the answer originated
![Screenshot_30](https://github.com/FlowiseAI/Flowise/assets/136782819/65cabcf4-fd1a-4981-b189-fd4dc60e4cf3)

**Comment by HenryHengZJ** [comment]
are you looking to get the sources or defining the priority of the tools?

for returning sources, its currently not supported on agent yet, you can use ConversationalRetrievalQAChain for that

**Reply by trepont** [reply]
Yes, I am looking to define the priority of the tools.
